### PR TITLE
fix: ternary-like expressions in github actions are weird. Also fix for missing rebase expressions

### PR DESCRIPTION
**Pull request by librarian**
In github actions there is a ternary-like operator, but it is not really [ternary](https://docs.github.com/en/actions/learn-github-actions/expressions#example).

>  It is important to note that the first value after the && must be truthy. Otherwise, the value after the || will always be returned.

What means truthy?

> Note that in conditionals, falsy values (false, 0, -0, "", '', null) are coerced to false and truthy (true and other non-falsy values) are coerced to true.

And bug occurs because 0 is casted, for some reason, to false

### CODE REVIEW

File: .github/workflows/build_and_test_cmake.yaml [hunk 1]
```diff
@@ -53,12 +53,18 @@ jobs:
       with:
         submodules: true
         ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: ${{ !contains(github.event.pull_request.labels.*.name, 'rebase') && 1 || 0 }}
+    - name: Rebase PR
```
Comment (budevg) [inline .github/workflows/build_and_test_cmake.yaml:57]: looks like this commit not just about fixing ternary operator but also about doing rebase ... better to split into separate commits

File: .github/workflows/build_and_test_cmake.yaml [hunk 2]
```diff
@@ -53,12 +53,18 @@ jobs:
...
+      if: ${{ github.event.pull_request.head.sha != '' && contains(github.event.pull_request.labels.*.name, 'rebase') }}
+      shell: bash
```
Comment (budevg) [inline .github/workflows/build_and_test_cmake.yaml:59]: Is re-basing during PR supposed to protect us from merging code which was not tested after rebase to latest main branch ? It won't help if all tests pass but user will decide to press the merge button after several days.

We could enforce `--ff-only` merges. Not sure if github supports it out of the box

**Review by budevg — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test_cmake.yaml (hunk 2 above)
Comment (librarian) [reply .github/workflows/build_and_test_cmake.yaml:59]: Idea is that it will lover chances of merging such code. We may implement on: push checks later

File: .github/workflows/build_and_test_cmake.yaml (hunk 1 above)
Comment (librarian) [reply .github/workflows/build_and_test_cmake.yaml:57]: This is just fixing places where rebase should happen, but was missed in previous iteration. It will work weird if you split it in separate changes.

Comment (librarian) [reply .github/workflows/build_and_test_cmake.yaml:57]: Do you mean move this thing to separate commit inside this PR? It will be squashed afterwards though

Comment (budevg) [reply .github/workflows/build_and_test_cmake.yaml:57]: maybe just  change commit name to `adding missing rebase and fixing ternary expressions` or something when you merge so it will be less confusing.
In general you don't have to squash you can "rebase and merge" and preserve two commits if it makes sense to preserve them. Anyway I don't insist on changing this PR. But would be great to have separate commits in the future

**Review by budevg — APPROVED** [review]
